bookstore/src/app/shared/services/store.service.ts

```diff
@@ -105,66 +108,76 @@ export class StoreService {
         return this.filterBooks(books, filterData);
       })
     );
-
   }
+
   /*
   * If your data set is too big this may raise performance issue.
   * You should implement server side filtering instead.
   */
   private filterBooks(books: Book[], filterData: any): Observable<Book[]> {
-    const filteredBooks = books.filter(p => {
+    const filteredBooks = books.filter(book => {
       const match = {
         search: false,
         genre: false,
         price: false,
         rating: false
       };
       // Search
-      if (
-        !filterData.search
-        || p.title.toLowerCase().indexOf(filterData.search.toLowerCase()) > -1
-        || p.description.indexOf(filterData.search) > -1
-        || p.tags.indexOf(filterData.search) > -1
-      ) {
+      if (!filterData.search || book.title.toLowerCase().indexOf(filterData.search.toLowerCase()) > -1 || book.description.indexOf(filterData.search) > -1 || book.tags.indexOf(filterData.search) > -1) {
         match.search = true;
       } else {
         match.search = false;
       }
       // Category filter
-      if (
-        p.genres.includes(filterData.genre)
-        || !filterData.genre
-        || filterData.genre === 'all'
-      ) {
+      if (book.genres.includes(filterData.genre) || !filterData.genre || filterData.genre === 'all') {
         match.genre = true;
       } else {
         match.genre = false;
       }
       // Price filter
-      if (
-        p.price.amount >= filterData.minPrice
-        && p.price.amount <= filterData.maxPrice
-      ) {
+      if (book.price.amount >= filterData.minPrice && book.price.amount <= filterData.maxPrice) {
         match.price = true;
       } else {
         match.price = false;
       }
       // Rating filter
-      if (
-        p.ratings.rating >= filterData.minRating
-        && p.ratings.rating <= filterData.maxRating
-      ) {
+      if (book.ratings.rating >= filterData.minRating && book.ratings.rating <= filterData.maxRating) {
         match.rating = true;
       } else {
         match.rating = false;
       }
 
       for (const m in match) {
-        if (!match[m]) { return false; }
+        if (!match[m]) { 
+          return false; 
+        }
       }
 
       return true;
     });
+    
     return of(filteredBooks);
   }
-}
+
+  public getOrders(): Observable<Order[]> {
+    const ordersDB = new OrdersDB();
+    return of(ordersDB.orders)
+    .pipe(
+      delay(500), 

```

Comment:
Can we make this delay value configurable?

Reply:
This `delay()` is only for mock data, once we complete actual service integration this will be removed.